
message RunTaskReq {
  string name = 1;

}



message Event {
    string server_id = 1; 
    string task_name = 2;
    oneof event {
        TaskStarted task_started = 3;
        TaskCompleted task_completed = 4;
        TaskFailed task_failed = 5;
    }
}



message Task {
  string server_id = 1;
  string name = 1;
  repeated string upstream_tasks = 3;
  repeated string downstream_tasks = 4;
}




message RegisterTaskServerReq {
    string name = 1;


}
